src/backend/api/reservations.js

```diff
@@ -0,0 +1,151 @@
+const express = require('express')
+const router = express.Router()
+const knex = require('../database')
+
+//GET. Returns all reservations
+router.get('/', async (request, response) => {
+  try {
+    const reservations = await knex('reservation')
+    response.json(reservations)
+  } catch (error) {
+    throw error
+  }
+})
+
+//GET. Returns reservation by id
+router.get('/:id', async (request, response) => {
+  try {
+    const reservationId = Number(request.params.id)
+    if (isNaN(reservationId)) {
+      response.status(400).json({
+        error: "Reservation wasn't found.",
+        hint: 'Id should be a number',
+      })
+      return
+    } else {
+      const reservations = await knex('reservation')
+      const isFoundId = reservations.some(
+        (reservation) => reservation.id === reservationId
+      )
+      if (!isFoundId) {
+        response.status(400).json({
+          error: "Reservation wasn't found.",
+          hint: 'Reservation with this id does not exist',
+        })
+        return
+      }
+      const reservationById = await knex('meal')
+        .select('*')
+        .where({ id: reservationId })
+      response.json(reservationById)
+    }
+  } catch (error) {
+    throw error
+  }
+})
+
+//POST.	Adds a new reservation
+router.post('/', async (request, response) => {
+  try {
+    const context = request.body
+    if (Object.keys(context).length === 0 || context === {}) {
+      response.statusCode(422).json({ message: 'Bad input' })
+      return
+    }
+    if (context.number_of_guests === '') {
+      response.json({ message: 'Please, provide number of guests' })
+      return
+    }
+    if (context.created_date === '') {
+      response.json({ message: 'Please, provide date' })
+      return
+    }
+    if (context.contact_phonenumber === '') {
+      response.json({ message: 'Please, provide phonenumber' })
+      return
+    }
+    if (context.contact_name === '') {
+      response.json({ message: 'Please, provide name' })
+      return
+    }
+    if (context.contact_email === '') {
+      response.json({ message: 'Please, provide email' })
+      return
+    }
+    if (context.meal_id === '') {
+      response.json({ message: 'Please, provide meal id' })
+      return
+    }
```

Comment:
Same comment about cleaning this up in a list instead of individual if clauses

Reply:
changed